### ISSUE TITLE
Endpoint validation does not match RFC URL standard

### ISSUE THREAD

**Issue by JohannesZahn** [DefectDojo/django-DefectDojo#5048]
**Bug description**
The Host validation regex found [here](https://github.com/DefectDojo/django-DefectDojo/blob/00b425742b783ada0f432241c2812ac1257feb73/dojo/models.py#L1199) does not support all valid host names according to [RFC3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.2). For example using the host  "_invalid._host.com" will not pass the regex even tho it should. See the Screenshots section to see why I think it should pass.

This leads to other problems like internal server errors when using the reimport api to import findings with these kinds of endpoints (starting with charakters other than [A-Za-z0-9]).

**Steps to reproduce**
1. Use the reimport API to import a findings with the endpoint "http://_invalid._host.com"

**Expected behavior**
The validation should not fail

**Deployment method**
Docker

**Screenshots** (optional)
![image](https://user-images.githubusercontent.com/28805778/132311355-f45e279e-6d9b-4859-9a84-f36e6d914567.png)
![image](https://user-images.githubusercontent.com/28805778/132311377-314b7ccd-63b4-4f46-871c-3a8d29e05293.png)
![image](https://user-images.githubusercontent.com/28805778/132311396-538a7708-4050-48a6-a56d-0d14b3ccbaa5.png)

**Console logs** (optional)
These logs were supplied by @EndPositive in Issue secureCodeBox/secureCodeBox#618
```
[30/Aug/2021 20:17:01] ERROR [django.request:224] Internal Server Error: /api/v2/reimport-scan/
Traceback (most recent call last):
  File "/app/./dojo/importers/reimporter/reimporter.py", line 162, in process_parsed_findings
    endpoint.clean()
  File "/app/./dojo/models.py", line 1218, in clean
    raise ValidationError(errors)
django.core.exceptions.ValidationError: ['Host "_invalid._host.com" has invalid format']
[...]
```

**Comment by damiencarol** [comment]
@JohannesZahn I'm concerned by the fact that fake data go to our API. Why using fake host names? Should it be more simple to not add a host as this attribute is not mandatory?

**Comment by JohannesZahn** [comment]
urls with leading underscores in the hostname are appereantly produced in some amass scans that we simply pipe through to the defectdojo reimport api which parses the urls into protocol, host, etc.. I am not sure wether the fake data is produced by the amass scan or was simply changed by @EndPositive to protect the real data.

**Comment by damiencarol** [comment]
ok. thanks for the feedback. Could you investigate? I'm curious about this one.
Anyway thanks to @kiblik a fix is coming ( #5049 ) .

**Comment by JohannesZahn** [comment]
@kiblik you mentioned you were following a rule when you specifically excluded underscores. I understand now why you did that. 
Part of a URL is...
> A host subcomponent, consisting of either a registered name (including but not limited to a hostname), or an IP address.
-https://en.wikipedia.org/wiki/URL#Syntax

Hostnames which are standardized in [RFC1123](https://datatracker.ietf.org/doc/html/rfc1123) may not contain underscores. However the host part of a URL is not limited to these standardized hostnames.

**Comment by JohannesZahn** [comment]
@damiencarol I will investigate and get back to you regarding the fake data

**Comment by EndPositive** [comment]
Hey everyone, the host was indeed fabricated. I should've used an example closer to the actual host, such as:

`_sip._tcp.domain.com`.

My mistake is that the domain part also contained an underscore (`_invalid._host.com`). I think @JohannesZahn is right about the validity of underscores in hostnames. Here's a [StackOverflow post](https://stackoverflow.com/a/2183140) with some interesting discussion on the topic. Also [this post](https://superuser.com/questions/59093/difference-between-host-name-and-domain-name) about hostname vs domains. I.e. domains **may not** contain underscores, hostnames **may**.

**Comment by damiencarol** [comment]
Awesome feedback guys! thanks.

**Comment by damiencarol** [comment]
Fixed with #5049
